File: app/src/main/java/app/fit/fitndflow/ui/features/categories/ConfirmationDialogFragment.java
package app.fit.fitndflow.ui.features.categories;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;

import com.fit.fitndflow.R;
import com.fit.fitndflow.databinding.DialogConfirmationFragmentBinding;

import app.fit.fitndflow.ui.features.common.CommonDialogFragment;


public class ConfirmationDialogFragment extends CommonDialogFragment {
    public static final int DELETE_CATEGORY = 1;
    public static final int DELETE_EXERCISE = 2;
    private static final String KEY_CALLBACK = "dialogCallback";
    private static final String KEY_TYPE = "deleteType";
    private static final String KEY_ID = "id";
    private DialogConfirmationFragmentBinding binding;
    private DialogCallbackDelete dialogCallbackDelete;
    private int deleteType;
    private int id;
    public static String TAG = "ConfirmationDialog";

    @Override
    protected int getBorderType() {
        return BORDERLINE;
    }

    @Override
    public ViewBinding getBinding() {
        binding = DialogConfirmationFragmentBinding.inflate(getLayoutInflater());
        return binding;
    }

    public static ConfirmationDialogFragment newInstance(DialogCallbackDelete dialogCallbackDelete, int deleteType, int id){
        ConfirmationDialogFragment confirmationDialogFragment = new ConfirmationDialogFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable(KEY_CALLBACK, dialogCallbackDelete);
        bundle.putInt(KEY_TYPE, deleteType);
        bundle.putInt(KEY_ID, id);
        confirmationDialogFragment.setArguments(bundle);

        return  confirmationDialogFragment;
    }

    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = super.onCreateView(inflater, container, savedInstanceState);
        Bundle bundle = getArguments();
        if (bundle != null) {
            dialogCallbackDelete = (DialogCallbackDelete) bundle.getSerializable(KEY_CALLBACK);
            deleteType = bundle.getInt(KEY_TYPE);
            id = bundle.getInt(KEY_ID);
        }
        initListeners();
        return view;
    }

    private void initListeners(){
        if (deleteType == DELETE_CATEGORY){
            binding.confirmationDialogTitle.setText(R.string.dialog_category_delete);
            binding.confirmationDialogAcceptBtn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    dialogCallbackDelete.onClickAcceptDelete(id);
                    dismissAllowingStateLoss();
                }
            });
        } else {
            binding.confirmationDialogTitle.setText(R.string.dialog_exercise_delete);
            binding.confirmationDialogAcceptBtn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    dialogCallbackDelete.onClickAcceptDelete(id);
                    dismissAllowingStateLoss();
                }
            });
        }
        binding.confirmationDialogCancelBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dismissAllowingStateLoss();
            }
        });
    }
}

File: app/src/main/java/app/fit/fitndflow/ui/features/categories/ConfirmationDialogFragment.kt
package app.fit.fitndflow.ui.features.categories

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import app.fit.fitndflow.ui.features.common.CommonDialogFragment
import com.fit.fitndflow.R
import com.fit.fitndflow.databinding.DialogConfirmationFragmentBinding

class ConfirmationDialogFragment : CommonDialogFragment() {

    companion object {
        @JvmField
        val DELETE_CATEGORY = 1
        @JvmField
        val DELETE_EXERCISE = 2
        @JvmField
        val DELETE_SERIE = 3
        @JvmField
        val TAG = "ConfirmationDialog"
        private val KEY_CALLBACK = "dialogCallback"
        private val KEY_TYPE = "deleteType"
        private val KEY_ID = "id"

        @JvmStatic
        fun newInstance(
            dialogCallbackDelete: DialogCallbackDelete,
            deleteType: Int,
            id: Int
        ): ConfirmationDialogFragment {
            val confirmationDialogFragment = ConfirmationDialogFragment()
            val bundle = Bundle()
            bundle.putSerializable(KEY_CALLBACK, dialogCallbackDelete)
            bundle.putInt(KEY_TYPE, deleteType)
            bundle.putInt(KEY_ID, id)
            confirmationDialogFragment.arguments = bundle
            return confirmationDialogFragment
        }
    }

    private lateinit var binding: DialogConfirmationFragmentBinding
    private val dialogCallbackDelete: DialogCallbackDelete by lazy {
        requireArguments().getSerializable(KEY_CALLBACK) as DialogCallbackDelete
    }
    private val deleteType: Int by lazy { requireArguments().getSerializable(KEY_TYPE) as Int }
    private val itemId: Int by lazy { requireArguments().getSerializable(KEY_ID) as Int }
    override fun getBorderType(): Int {
        return BORDERLINE
    }

    override fun getBinding(): ViewBinding {
        binding = DialogConfirmationFragmentBinding.inflate(layoutInflater)
        return binding
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = super.onCreateView(inflater, container, savedInstanceState)
        initListeners()
        return view
    }

    private fun initListeners() {
        binding.apply {
            when (deleteType) {
                DELETE_CATEGORY -> {
                    confirmationDialogTitle.setText(R.string.dialog_category_delete)
                    confirmationDialogAcceptBtn.setOnClickListener {
                        dialogCallbackDelete.onClickAcceptDelete(itemId)
                        dismissAllowingStateLoss()
                    }
                }

                DELETE_EXERCISE -> {
                    confirmationDialogTitle.setText(R.string.dialog_exercise_delete)
                    confirmationDialogAcceptBtn.setOnClickListener {
                        dialogCallbackDelete.onClickAcceptDelete(itemId)
                        dismissAllowingStateLoss()
                    }
                }

                else -> {
                    confirmationDialogTitle.setText(R.string.dialog_serie_delete)
                    confirmationDialogAcceptBtn.setOnClickListener {
                        dialogCallbackDelete.onClickAcceptDelete(itemId)
                        dismissAllowingStateLoss()
                    }
                }
            }
            confirmationDialogCancelBtn.setOnClickListener {
                dismissAllowingStateLoss()
            }
        }
    }
}